src/test/java/peoplesoft/model/employment/EmploymentTest.java

```diff
@@ -0,0 +1,179 @@
+
+        // Person missing
+        assertThrows(EmploymentNotFoundException.class, () -> employment.disassociate(EATING, BOB));
+    }
+
+    @Test
+    public void deletePerson_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Employment().deletePerson(null));
+    }
+
+    @Test
+    public void deletePerson_success() {
+        Employment employment = new Employment();
+        employment.associate(EATING, ALICE);
+        employment.associate(EATING, BOB);
+        employment.associate(RUNNING, ALICE);
+
+        // Delete Bob
+        employment.deletePerson(BOB);
+        assertTrue(employment.getAllJobs().containsKey(EATING.getJobId()));
+        assertTrue(employment.getAllJobs().get(EATING.getJobId()).contains(ALICE.getPersonId()));
+        assertFalse(employment.getAllJobs().get(EATING.getJobId()).contains(BOB.getPersonId()));
+
+        // Delete Alice
+        employment.deletePerson(ALICE);
+        assertFalse(employment.getAllJobs().containsKey(EATING.getJobId()));
+        assertFalse(employment.getAllJobs().containsKey(RUNNING.getJobId()));
+    }
+
+    @Test
+    public void editPerson_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Employment().editPerson(null, ALICE));
+        assertThrows(NullPointerException.class, () -> new Employment().editPerson(ALICE, null));
+    }
+
+    @Test
+    public void editPerson_success() {
+        Employment employment = new Employment();
+        employment.associate(EATING, ALICE);
+        employment.associate(RUNNING, ALICE);
+
+        // Edit Alice to Bob
+        employment.editPerson(ALICE, BOB);
+        assertTrue(employment.getAllJobs().containsKey(EATING.getJobId()));
+        assertFalse(employment.getAllJobs().get(EATING.getJobId()).contains(ALICE.getPersonId()));
+        assertTrue(employment.getAllJobs().get(EATING.getJobId()).contains(BOB.getPersonId()));
+        assertTrue(employment.getAllJobs().containsKey(RUNNING.getJobId()));
+        assertFalse(employment.getAllJobs().get(RUNNING.getJobId()).contains(ALICE.getPersonId()));
+        assertTrue(employment.getAllJobs().get(RUNNING.getJobId()).contains(BOB.getPersonId()));
+    }
+
+    @Test
+    public void deleteJob_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Employment().deleteJob(null));
+    }
+
+    @Test
+    public void deleteJob_success() {
+        Employment employment = new Employment();
+        employment.associate(EATING, ALICE);
+        employment.associate(EATING, BOB);
+        employment.associate(RUNNING, ALICE);
+
+        // Delete eating
+        employment.deleteJob(EATING);
+        assertFalse(employment.getAllJobs().containsKey(EATING.getJobId()));
+        assertTrue(employment.getAllJobs().containsKey(RUNNING.getJobId()));
+
+        // Delete Alice
```

Comment:
?

Reply:
Should I add more comments to describe what is happening?

Reply:
more like, the comments don't seem to be describing what's happening here -- isn't the `RUNNING` job being deleted here instead